server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts

```diff
@@ -110,7 +110,22 @@ export default class ServiceProviderReferralsController {
     this.renderDashboardWithoutPagination(req, res, referralsSummary, 'My cases')
   }
 
+  private handlePaginatedSearchText(req: Request) {
+    if (req.query.paginated === 'true') {
+      req.body['case-search-text'] = req.session.searchText
```

Comment:
why we need to tamper the request body here. instead you can return the search text and assign it in line no 129.
```const searchText = handlePaginatedSearchText(params...)```